src/ansible_navigator/configuration_subsystem/definitions.py

```diff
@@ -134,16 +138,22 @@ class SubCommand(SimpleNamespace):
     epilog: Union[None, str] = None
 
 
-class ApplicationConfiguration(SimpleNamespace):
+@dataclass
+class ApplicationConfiguration:
+    # pylint: disable=too-many-instance-attributes
     """The main object for storing an application config"""
 
-    application_name: str = ""
-    entries: List[SettingsEntry] = []
+    application_version: str
+    entries: List[SettingsEntry]
     internals: SimpleNamespace
+    post_processor: "NavigatorPostProcessor"
```

Comment:
From a readability standpoint, I'd prefer just importing it unconditionally and not using a string for the type name.

Reply:
It results in a circular import because post process import from here.....

Reply:
what probably really should be done is NavigatorPostProcessor should have a base class defined here, but, that's for another day :)